src/main/webapp/app/course/competencies/prerequisite.service.ts

```diff
@@ -0,0 +1,78 @@
+import { Prerequisite, PrerequisiteResponseDTO } from 'app/entities/prerequisite.model';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable, map } from 'rxjs';
+import { CourseCompetency } from 'app/entities/competency.model';
+import { convertDateFromServer } from 'app/utils/date.utils';
+
+@Injectable({
+    providedIn: 'root',
+})
+export class PrerequisiteService {
+    private resourceURL = 'api';
+
+    constructor(private httpClient: HttpClient) {}
+
+    getAllPrerequisitesForCourse(courseId: number): Observable<Prerequisite[]> {
+        return this.httpClient.get<PrerequisiteResponseDTO[]>(`${this.resourceURL}/courses/${courseId}/competencies/prerequisites`, { observe: 'response' }).pipe(
+            map((resp) => {
+                if (!resp.body) {
+                    return [];
```

Comment:
How can there be no response body? Is there no exception from the server then?

Reply:
I just looked it up and you're right, the server always returns an array.